File: testdata/Mock.swift
// Swift test fixture for hollowcheck - mock data
import Foundation

/// MockUser represents a user with fake data.
struct MockUser {
    var id: String
    var email: String
    var name: String
}

/// MockConfig has placeholder values.
struct MockConfig {
    var apiKey: String
    var endpoint: String
    var password: String
}

/// GetTestUser returns a user with mock data.
func getTestUser() -> MockUser {
    return MockUser(
        id: "12345",
        email: "test@example.com",
        name: "foo"
    )
}

/// GetAnotherUser returns another mock user.
func getAnotherUser() -> MockUser {
    return MockUser(
        id: "00000",
        email: "user@example.com",
        name: "bar"
    )
}

/// GetMockConfig returns a config with fake data.
func getMockConfig() -> MockConfig {
    return MockConfig(
        apiKey: "asdf1234",
        endpoint: "https://api.example.com/v1",
        password: "changeme"
    )
}

/// GetDescription returns lorem ipsum placeholder text.
func getDescription() -> String {
    return "lorem ipsum dolor sit amet"
}

/// GetPhoneNumber returns a placeholder phone.
func getPhoneNumber() -> String {
    return "xxx-xxx-xxxx"
}

/// SequentialIDs returns fake sequential IDs.
func sequentialIDs() -> [String] {
    return ["11111", "22222", "33333"]
}

File: testdata/Mock.scala
// Scala test fixture for hollowcheck - mock data
package testdata

/** MockUser represents a user with fake data. */
case class MockUser(id: String, email: String, name: String)

/** MockConfig has placeholder values. */
case class MockConfig(apiKey: String, endpoint: String, password: String)

object Mock {
  /** GetTestUser returns a user with mock data. */
  def getTestUser(): MockUser = {
    MockUser(
      id = "12345",
      email = "test@example.com",
      name = "foo"
    )
  }

  /** GetAnotherUser returns another mock user. */
  def getAnotherUser(): MockUser = {
    MockUser(
      id = "00000",
      email = "user@example.com",
      name = "bar"
    )
  }

  /** GetMockConfig returns a config with fake data. */
  def getMockConfig(): MockConfig = {
    MockConfig(
      apiKey = "asdf1234",
      endpoint = "https://api.example.com/v1",
      password = "changeme"
    )
  }

  /** GetDescription returns lorem ipsum placeholder text. */
  def getDescription(): String = {
    "lorem ipsum dolor sit amet"
  }

  /** GetPhoneNumber returns a placeholder phone. */
  def getPhoneNumber(): String = {
    "xxx-xxx-xxxx"
  }

  /** SequentialIDs returns fake sequential IDs. */
  def sequentialIDs(): Seq[String] = {
    Seq("11111", "22222", "33333")
  }
}
